SConstruct

```diff
@@ -288,7 +288,7 @@ if selected_platform in platform_list:
     if (env["warnings"] == 'yes'):
         print("WARNING: warnings=yes is deprecated; assuming warnings=all")
 
-    if (os.name == "nt" and os.getenv("VSINSTALLDIR")): # MSVC, needs to stand out of course
+    if (os.name == "nt" and os.getenv("VSINSTALLDIR") and platform_arg != "javascript"): # MSVC, needs to stand out of course
```

Comment:
I think this should be `platform_arg == "windows" or platform_arg == "uwp"`. The warnings settings apply only to platforms actually built with MSVC.